src/ncp/spinel.c

```diff
@@ -2030,6 +2047,27 @@ main(void)
     }
 
     {
+        const char *str = NULL;
+
+        // Length ends right before the string.
+        len = spinel_datatype_unpack(buffer, 8, "CiiLU", NULL, NULL, NULL, NULL, &str);
+
+        if (len != -1)
+        {
+            printf("error:%d: len != -1; (%d)\n", __LINE__, (int)len);
+            goto bail;
+        }
+
+        if (str != NULL)
+        {
+            printf("error:%d: str != NULL\n", __LINE__);
+            goto bail;
+        }
+    }
+
+    len = 30;
```

Comment:
The `len = 30;` statement seems unnecessary. `len` is update in the next `{  ...  }`

Reply:
`len` is not updated before the next call to `spinel_datatype_unpack()`, which is why it is set here. Without this the test (appropriately) fails.

Reply:
Ah, I see. That's fine.